endpoints/registry_test.go

```diff
@@ -0,0 +1,46 @@
+package endpoints
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestEmptyRegistry(t *testing.T) {
+	r := NewRegistry(RegistryOptions{})
+	m := r.GetMetrics("some key")
+
+	assert.Equal(t, time.Time{}, m.DetectedTime())
+	assert.Equal(t, int64(0), m.InflightRequests())
+}
+
+func TestSetAndGet(t *testing.T) {
+	r := NewRegistry(RegistryOptions{})
+
+	mBefore := r.GetMetrics("some key")
+	r.IncInflightRequest("some key")
+	mAfter := r.GetMetrics("some key")
+
+	assert.Equal(t, int64(0), mBefore.InflightRequests())
+	assert.Equal(t, int64(1), mAfter.InflightRequests())
+
+	ts, _ := time.Parse("2006-01-02", "2023-08-29")
```

Comment:
Use predefined format constant for time.Parse, see https://pkg.go.dev/time#pkg-constants